src/iceberg/catalog/rest/types.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "iceberg/partition_spec.h"
+#include "iceberg/schema.h"
+#include "iceberg/sort_order.h"
+#include "iceberg/table_identifier.h"
+#include "iceberg/table_metadata.h"
+
+namespace iceberg::rest {
+
+struct ListNamespaceResponse {
+  std::vector<std::vector<std::string>> namespaces;
+  // TODO(Li Feiyang): Add next_page_token
+};
+
+struct CreateNamespaceRequest {
+  std::vector<std::string> namespaces;
+  std::optional<std::unordered_map<std::string, std::string>> properties;
+};
+
+struct CreateNamespaceResponse {
+  std::vector<std::string> namespaces;
+  std::optional<std::unordered_map<std::string, std::string>> properties;
+};
+
+struct GetNamespacePropertiesResponse {
+  std::vector<std::string> namespaces;
+  std::optional<std::unordered_map<std::string, std::string>> properties;
+};
+
+struct UpdateNamespacePropertiesRequest {
+  std::optional<std::vector<std::string>> removals;
+  std::optional<std::unordered_map<std::string, std::string>> updates;
+};
+
+struct UpdateNamespacePropsResponse {
+  std::vector<std::string> updated;
+  std::vector<std::string> removed;
+  std::optional<std::vector<std::string>> missing;
+};
+
+struct ListTableResponse {
+  std::optional<std::vector<TableIdentifier>> identifiers;
+  // TODO(Li Feiyang): Add next_page_token
+};
+
+struct CreateTableRequest {
+  std::string name;
+  std::optional<std::string> location;
```

Comment:
```suggestion
  std::string location;
```

An empty string is enough to represent an optional one.